### PR TITLE
fix: get helm values from adjacent apps + clean up empty nested maps

### PR DESCRIPTION
**Pull request by DavidSpek**
## Summary
After https://github.com/pluralsh/plural-cli/pull/301, the ability to get helm values from adjacent applications broke since it was still trying to get those from the `values.yaml` and not a combination of the `default-values.yaml` and `values.yaml`. This PR fixes this. Along the way, it occurred that empty nested maps could be left in the `values.yaml`, which has also been solved.

TODO: we still need to add unit tests for cleaning up the empty maps.

## Test Plan
- Manually test with the apps that were causing problems

## Checklist

- [x] If required, I have updated the Plural documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] I have added a meaningful title and summary to convey the impact of this PR to a user.
- [x] I have added relevant labels to this PR to help with categorization for release notes.

### CODE REVIEW

File: pkg/scaffold/application.go [hunk 1]
```diff
@@ -28,14 +29,35 @@ func NewApplications() (*Applications, error) {
...
+		if err := yaml.Unmarshal(defaultValsContent, &defaultVals); err != nil {
+			return res, err
+		}
+	}
 
-	err = yaml.Unmarshal(content, &res)
+	for k, v := range defaultVals {
+		res[k] = v
+	}
+	for k, v := range vals {
```
Comment (michaeljguarino) [inline pkg/scaffold/application.go:58]: isn't this merge technically too strict?  Like it wouldn't handle nested objects properly

Comment (zreigz) [reply pkg/scaffold/application.go:58]: it's not the typical merge. We have to override default values with values

Comment (DavidSpek) [reply pkg/scaffold/application.go]: Yeah I wasn't happy about that merge, but got caught up removing empty maps properly. Lukasz change should have fixed this merging.

@zreigz Wouldn't what we do in https://github.com/pluralsh/plural-cli/blob/main/pkg/scaffold/helm.go#L200 also be too strict then?

**Review by zreigz — APPROVED** [review]
lgtm

File: pkg/scaffold/application.go (hunk 1 above)
Comment (zreigz) [reply pkg/scaffold/application.go]: @DavidSpek  here is another problem, I will make another PR for this
